### PR TITLE
Wrap loading animation with condition of env variable

### PR DESCRIPTION
**Pull request by fxfactorial**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

Re: https://github.com/fastlane/fastlane/pull/11352#issuecomment-355043978

### CODE REVIEW

**Comment by fxfactorial** [comment]
I signed it!

File: fastlane/lib/fastlane/cli_tools_distributor.rb [hunk 1]
```diff
@@ -15,24 +15,27 @@ def running_help_command?
...
-        require "tty-spinner"
-        require_fastlane_spinner = TTY::Spinner.new("[:spinner] 🚀 ", format: :dots)
-        require_fastlane_spinner.auto_spin
-
-        # this might take a long time if there is no Gemfile :(
-        # That's why we show the loading indicator here also
-        require "fastlane"
-
-        require_fastlane_spinner.success
+        if !FastlaneCore::Env.truthy?("FASTLANE_DISABLE_ANIMATION")
```
Comment (hjanuschka) [inline fastlane/lib/fastlane/cli_tools_distributor.rb]: i think this is a chicken-egg problem, we haven't loaded fastlane_core at this point-in-code - however you are using `FastlaneCore::`- so i guess you'd need to check the env with raw ruby `ENV[]` style.

this also should fix tests, i hope!

**Review by hjanuschka — CHANGES_REQUESTED** [review]

**Comment by hjanuschka** [comment]
@fxfactorial congrats on your first PR 🚀  🎉   and welcome to the fastlane project.
please see my comments that need to be addressed, and once we have the ci green it should be good to be merged🐙

File: fastlane/lib/fastlane/cli_tools_distributor.rb (hunk 1 above)
Comment (fxfactorial) [reply fastlane/lib/fastlane/cli_tools_distributor.rb]: pushed, thanks.

Comment (KrauseFx) [reply fastlane/lib/fastlane/cli_tools_distributor.rb]: Nice catch @hjanuschka 👍

**Review by KrauseFx — COMMENTED** [review]
Awesome, thanks @fxfactorial. The startup time is one place where we show a loading indicator, and then for the rest of the code base, we're gonna use the `Helper` method

So we'll need a check over here as well: https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/helper.rb#L277-L284

The first line of each method could just be
```ruby
return if FastlaneCore::Env.truthy?("FASTLANE_DISABLE_ANIMATION")
```

**Comment by fxfactorial** [comment]
Okay, so I am not clear about what you want me to do now then? Is there anything for me to change since your linked file seems to imply it has already been done

**Comment by hjanuschka** [comment]
@fxfactorial i think @KrauseFx  means that in addition to your current changes, the check for the ENV should also be done here: https://github.com/fastlane/fastlane/blob/master/fastlane_core/lib/fastlane_core/helper.rb#L277

as you can assume that fastlane_core is already loaded at this point-in-code.
you can add 

```
return if FastlaneCore::Env.truthy?("FASTLANE_DISABLE_ANIMATION")
```

the `truthy?`  has some nice extras like `FASTLANE_DISABLE_ANIMATION=false` is not `truthy?`

**Comment by fxfactorial** [comment]
Cool, thanks @hjanuschka and I didn't know about that neat ruby syntax feature. Rebased and pushed.

**Comment by carlosefonseca** [comment]
Could `FASTLANE_DISABLE_ANIMATION` be set to true if `is_ci?` is true?

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 2]
```diff
@@ -275,6 +275,7 @@ def self.gem_path(gem_name)
 
     # Show/Hide loading indicator
     def self.show_loading_indicator(text = "🚀")
+      return if FastlaneCore::Env.truthy?("FASTLANE_DISABLE_ANIMATION")
       @require_fastlane_spinner = TTY::Spinner.new("[:spinner] #{text} ", format: :dots)
       @require_fastlane_spinner.auto_spin
     end
```
Comment (hjanuschka) [inline fastlane_core/lib/fastlane_core/helper.rb]: one last nitpick, `self.hide_loading_indicator`  also needs to be worked on other wise it crashes as @require_fastlane_spinner  might not be initialized.

**Review by hjanuschka — CHANGES_REQUESTED** [review]

File: fastlane_core/lib/fastlane_core/helper.rb (hunk 2 above)
Comment (fxfactorial) [reply fastlane_core/lib/fastlane_core/helper.rb]: okay, i think i did that just now.

**Comment by KrauseFx** [comment]
@fxfactorial hey, hey, sorry, totally my bad, I wasn't thinking. Even if not showing the loading indicator, we'll still have to show the text, as it might contain important information.

Sorry for the back and forth - I'm happy to do the extra work if you don't feel like updating the PR, I'd totally understand 👍

**Comment by fxfactorial** [comment]
Sure, np. If you're offering to continue it & finish it, then I'm happy about that.

**Comment by carlosefonseca** [comment]
@fxfactorial Maybe switch it to an `if` and add `UI.message(text)`.
Like what I was doing here https://github.com/fastlane/fastlane/pull/11442/files

**Comment by KrauseFx** [comment]
Hey everyone, sorry, couldn't push on your branch, and ended up in a new PR, you can find it here https://github.com/fastlane/fastlane/pull/11447

Thanks for your contribution @fxfactorial and @carlosefonseca 👍

**Pull request closed without merging**